arroyo/processing/strategies/run_task_with_multiprocessing.py

```diff
@@ -839,14 +839,19 @@ def submit(
             # compression.)
             self.__batch_builder.append(message)
 
-    def close(self) -> None:
+    def _do_close(self) -> None:
         self.__closed = True
 
+        signal.signal(signal.SIGCHLD, self.original_sigchld)
```

Comment:
I am not sure I get the reason to reinstall the original sigchld handler inside the close method.

Reply:
as written now if this is used in a process (say, getsentry's testsuite) the SIGCHILD handler persists after

in my case it's causing test pollution by a future test being broken by the sigchld handler